pseudopotential_calculator/calculations/slab.py

```diff
@@ -26,36 +23,100 @@
     from matplotlib.lines import Line2D
 
 
-class SlabOptimizationParams(OptimizationParamsBase):
-    """Parameters of a slab calculation."""
+def _get_sorted_heights(atom: Atoms) -> np.ndarray[Any, np.dtype[np.float64]]:
+    heights = cast(np.ndarray[Any, np.dtype[np.float64]], atom.positions[:, 2])  # type: ignore bad lib
+    return np.unique(heights)[::-1]
 
-    @property
-    def kpoint_mp_grid(self: Self) -> str:
-        return f"{self.n_k_points} {self.n_k_points} {1}"
 
+def get_height_of_layer(atoms: Atoms, layer_idx: int) -> float:
```

Comment:
just idx not layer_idx (layer is implied by the function name), try to rename it consistently everywhere